services/QuillLMS/app/services/classlink_integration/school_persister.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module ClasslinkIntegration
+  class SchoolPersister < ApplicationService
+    attr_reader :name, :district_id, :nces_id
+
+    def initialize(name:, district_id:, nces_id: nil)
+      @name = name
+      @district_id = district_id
+      @nces_id = nces_id
+    end
+
+    def run
+      if nces_id
+        School.find_or_create_by!(nces_id:) do |school|
+          school.name = name
+          school.district_id = district_id
+        end
+      else
+        School.find_or_create_by!(name:, district_id:)
+      end
+    end
```

Comment:
There may be plans to extend this later, in which case my suggestion won't make sense, but I prefer early returns over `if ... else` blocks.  Take or leave this, but I'd probably write it as:
```ruby
def run
  return School.find_or_create_by!(name:, district_id:) unless nces_id

  School.find_or_create_by!(nces_id:) do |school|
     school.name = name
     school.district_id = district_id
   end
end
```